mousestyles/ultradian/__init__.py

```diff
@@ -0,0 +1,358 @@
+
+    Examples
+    --------
+    >>> res = seasonal_decomposition(strain=0, mouse=0, feature="W",
+                                     bin_width=30, period_length = 24)
+    """
+    freq = int(period_length * 60 / bin_width)
+    ts = aggregate_interval(strain=strain, mouse=mouse,
+                            feature=feature, bin_width=bin_width)
+    res = sm.tsa.seasonal_decompose(ts.values, freq=freq, model="additive")
+    return(res)
+
+
+def strain_seasonal(strain, mouse, feature, bin_width, period_length):
+    """
+    Use seansonal decomposition model on the time series
+    of specified strain, mouse, feature and bin_width.
+    return the seasonal term and the plot of seasonal term
+    by mouse of a set of mouses in a strain
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: list, set or tuple
+        nonnegative integer indicating the mouse number
+    feature: {"AS", "F", "IS", "M_AS", "M_IS", "W"}
+    bin_width: int
+        number of minutes, the time interval for data aggregation
+    period_length: int
+        number of hours, usually the significant period
+        length indicated by Lomb-scargle model
+
+    Returns
+    -------
+    seasonal_all: numpy array containing the seasonal term for every
+        mouse indicated by the input parameter
+    seasonal_plot: plot of seasonal term by mouse
+
+    Examples
+    --------
+    >>> res = strain_seasonal(strain=0, mouse={0, 1, 2, 3}, feature="W",
+                              bin_width=30, period_length = 24)
+    """
+    # seasonal decomposition
+    seasonal_all = np.array([])
+    freq = int(period_length * 60 / bin_width)
+    for m in mouse:
+        res = seasonal_decomposition(
+            strain, m, feature, bin_width, period_length)
+        seasonal_all = np.append(seasonal_all, res.seasonal[0:freq])
+    seasonal_all = seasonal_all.reshape([len(mouse), -1])
+    # seasonal plot
+    seasonal_plot = plt.figure()
+    ax = seasonal_plot.add_subplot(1, 1, 1)
+    time = np.arange(0, period_length, bin_width / 60)
+    for i in np.arange(len(mouse)):
+        ax.plot(time, seasonal_all[i, :])
+    ax.legend(['mouse' + str(i) for i in np.arange(len(mouse))],
+              loc='upper right', prop={'size': 8})
+    plt.xlabel('Time')
+    plt.xlabel('Time')
+    plt.ylabel('Seasonal Term')
+    return seasonal_all, seasonal_plot
+
+
+def mix_strain(data, feature):
+    """
+    Fit the linear mixed model onto our aggregate data. The fixed effects
+    are the hour, strain, interactions between hour and strain; The random
+    effect is mouse because we want to make sure that the different mouses
+    will not give out any differences. We added two dummy variables:
+    strain0 and strain1 to be our fixed effects.
+
+    Parameters
+    ----------
+        data: data frame output from aggregate_data function
+        feature: {"AS", "F", "IS", "M_AS", "M_IS", "W"}
```

Comment:
Same problem as in function `seasonal_decomposition`, we are supposed to add the seasonal decomposition analysis for travel distance data.